themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,196 @@
-            </div>
-        );
-    }
-};
\ No newline at end of file
+
+export interface ApplicationsPageState {
+  isRowOpen: boolean[];
+  applications: Application[];
+}
+
+export interface GrantedScope {
+  displayTest: string;
+  id: string;
+  name: string;
+}
+
+export interface Consent {
+  createDate: number;
+  grantedScopes: GrantedScope[];
+  lastUpdatedDate: number;
+}
+
+interface Application {
+  baseUrl: string;
+  clientId: string;
+  clientName: string;
+  consent: Consent;
+  description: string;
+  inUse: boolean;
+  offlineAccess: boolean;
+  userConsentRequired: boolean;
+  scope: string[];
+}
+
+export class ApplicationsPage extends React.Component<ApplicationsPageProps, ApplicationsPageState> {
+
+  public constructor(props: ApplicationsPageProps) {
+    super(props);
+    this.state = {
+      isRowOpen: [],
+      applications: []
+    };
+
+    this.fetchApplications();
+  }
+
+  private removeConsent = (clientId: string) => {
+    AccountServiceClient.Instance.doDelete("/applications/" + clientId + "/consent")
+      .then(() => {
+        this.fetchApplications();
+      });
+  }
+
+  private onToggle = (row: number): void => {
+    const newIsRowOpen: boolean[] = this.state.isRowOpen;
+    newIsRowOpen[row] = !newIsRowOpen[row];
+    this.setState({ isRowOpen: newIsRowOpen });
+  };
+
+  private fetchApplications(): void {
+    AccountServiceClient.Instance.doGet("/applications")
+      .then((response: AxiosResponse<Object>) => {
+        const applications = response.data as Application[];
+        this.setState({
+          isRowOpen: this.collapseRows(applications),
+          applications: applications
+        });
+      });
+  }
+
+  private collapseRows(applications: Application[]): boolean[] {
+    const openRows: boolean[] = new Array<boolean>().fill(true);
+    applications.forEach((application: Application, appIndex: number) => {
+      openRows[appIndex] = false;
+    });
+    return openRows;
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <ContentPage title="applications">
+        <DataList id="applications-list" aria-label="Applications">
+          {this.state.applications.map((application: Application, appIndex: number) => {
+            return (
+              <DataListItem key={'application-' + appIndex} aria-labelledby="applications-list" isExpanded={this.state.isRowOpen[appIndex]}>
+                <DataListItemRow>
+                  <DataListToggle
+                    onClick={() => this.onToggle(appIndex)}
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="expandable"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell width={2} key={'app-' + appIndex}>
+                        <BuilderImageIcon /> <Text component={TextVariants.small}>{application.clientId}</Text>
```

Comment:
I think we should display `clientName` instead of just `clientId` if it's not null.

Reply:
Also, is there any reason why the clientId is smaller than rest of the fields on the same row?

Reply:
@vmuzikar changed to name with clientId as fallback. Stian suggested this some time ago: https://lists.jboss.org/pipermail/keycloak-dev/2019-October/012698.html

@vmuzikar regards the font-size, no specific reason. Could you please tell me which one you think would be better: https://patternfly-react.surge.sh/patternfly-4/documentation/react/components/text ?

Reply:
@abstractj Let me answer you with a question. :) Do we need to put client name/ID in `<Text>` component at all? We don't put e.g. "Internal" to this component and it looks good without it.

Reply:
@vmuzikar I can try to remove it and see how it looks this afternoon.